pkg/controllers/workgenerator/controller.go

```diff
@@ -874,23 +875,56 @@ func (r *Reconciler) SetupWithManager(mgr controllerruntime.Manager) error {
+				newAvailableCondition := meta.FindStatusCondition(newWork.Status.Conditions, fleetv1beta1.WorkConditionTypeAvailable)
+
+				// we try to filter out events, we only need to handle the updated event if the applied or available condition flip between true and false
+				// or the failed placements are changed.
+				if condition.EqualCondition(oldAppliedCondition, newAppliedCondition) && condition.EqualCondition(oldAvailableCondition, newAvailableCondition) {
+					if !condition.IsConditionStatusTrue(newAppliedCondition, newWork.Generation) || !condition.IsConditionStatusTrue(newAvailableCondition, newWork.Generation) {
+						// we need to compare the failed placement if the work is not applied or available
+						oldFailedPlacements := extractFailedResourcePlacementsFromWork(oldWork)
+						newFailedPlacements := extractFailedResourcePlacementsFromWork(newWork)
+						if isFailedResourcePlacementsEqual(oldFailedPlacements, newFailedPlacements) {
+							klog.V(2).InfoS("The failed placement list didn't change on failed work, no need to reconcile", "oldWork", klog.KObj(oldWork), "newWork", klog.KObj(newWork))
+							return
+						}
+					} else {
+						klog.V(2).InfoS("The work applied or available condition stayed as true, no need to reconcile", "oldWork", klog.KObj(oldWork), "newWork", klog.KObj(newWork))
+						return
+					}
 				}
-				klog.V(2).InfoS("Received a work update event", "work", klog.KObj(newWork), "parentBindingName", parentBindingName)
 				// We need to update the binding status in this case
+				klog.V(2).InfoS("Received a work update event that we need to handle", "work", klog.KObj(newWork), "parentBindingName", parentBindingName)
 				queue.Add(reconcile.Request{NamespacedName: types.NamespacedName{
 					Name: parentBindingName,
 				}})
 			},
 		}).
 		Complete(r)
 }
+
+func isFailedResourcePlacementsEqual(oldFailedResourcePlacements, newFailedResourcePlacements []fleetv1beta1.FailedResourcePlacement) bool {
+	if len(oldFailedResourcePlacements) != len(newFailedResourcePlacements) {
+		return false
+	}
+	sort.Slice(oldFailedResourcePlacements, func(i, j int) bool {
+		return condition.LessFuncFailedResourcePlacements(oldFailedResourcePlacements[i], oldFailedResourcePlacements[j])
+	})
+	sort.Slice(newFailedResourcePlacements, func(i, j int) bool {
+		return condition.LessFuncFailedResourcePlacements(newFailedResourcePlacements[i], newFailedResourcePlacements[j])
+	})
+	for i := range oldFailedResourcePlacements {
+		oldFailedResourcePlacement := oldFailedResourcePlacements[i]
+		newFailedResourcePlacement := newFailedResourcePlacements[i]
+		if !apiequality.Semantic.DeepEqual(oldFailedResourcePlacement.ResourceIdentifier, newFailedResourcePlacement.ResourceIdentifier) {
+			return false
+		}
+		if !condition.EqualCondition(&oldFailedResourcePlacement.Condition, &newFailedResourcePlacement.Condition) {
```

Comment:
the old observed generation will be less than or equal to the new generation but the equalCondition code is flipped.